app/Services/GuardianService.ts

```
@@ -162,6 +162,22 @@ export default class GuardianService {
     }
   }
 
+  public async getPolicyId(policyName: string): Promise<string | null> {
+    let policyId: string;
+    Sentry.addBreadcrumb({ message: `Getting policy ID for policyName: ${policyName}`, level });
+
+    try {
+      const { data: policies } = await this.axios.get('policies');
+      const policy = policies.find(({ name }) => name === policyName);
+      policyId = policy?.id ? policy.id : null;
+    } catch (error) {
+      Sentry.captureException(error);
+      throw new GuardianException('Error occurred retrieving policy ID', StatusCodes.InternalServerError);
+    }
+
+    return policyId;
```

As above. You can just go

```
try {
   ....
   return policy?.id;
}
```

I can't quite think through the logic of that, but essentially you don't really need the ternary. I think you can null coalesce but you don't want to return an undefined instead of a null so you might need to null coalesce to null, which is... weird. 

Moreover though I can't help thinking that this should actually through an exception. If you're asking for a policy id `null` isn't really a valid answer. Maybe it should throw an exception that there is no policy found?

---

Yup agree, should throw a no policy found exception.
Otherwise, that will save policyId as null and throw in letter stage of using calls with policy.